### ISSUE TITLE
Crash in Exiv2 when loading files in parallel

### ISSUE THREAD

**Issue by Ribtoks** [Ribtoks/xpiks#346]
code: 0xe06d7363

```
Locals      
    childName   0x234e9a82380 "xmp:Rating"  char *
    childOptions    0   unsigned long
    isArrayItem false   bool
    isTopLevel  true    bool
    isValueNode false   bool
    newChild    0xcccccccccccccccc  XMP_Node *
    value   0x7ff8ed1b7625 ""   char *
    xmlNode 0x234e9a82340   XML_Node *
    xmpParent   0x234e97d3a90   XMP_Node *
```

[full_stacktrace.txt](https://github.com/Ribtoks/xpiks/files/486748/full_stacktrace.txt)

```
> # 41  Id: 3b00.3e7c Suspend: 1 Teb: 00000091`afeef000 Unfrozen "QThread"
> Child-SP          RetAddr           Call Site
> 00000091`b2efc690 00007ff9`23e481c0 KERNELBASE!RaiseException+0x68
> 00000091`b2efc770 00007ff8`ed02b167 VCRUNTIME140D!_CxxThrowException(void * pExceptionObject = 0x00000091`b2efc8d0, struct _s__ThrowInfo * pThrowInfo = 0x00007ff8`ed223f88)+0x130 [f:\dd\vctools\crt\vcruntime\src\eh\throw.cpp @ 136]
> 00000091`b2efc820 00007ff8`ed029f3f libexiv2!AddChildNode(class XMP_Node * xmpParent = 0x000001db`ce4f1ce0, class XML_Node * xmlNode = 0x000001db`ce3d7f20, char * value = 0x00007ff8`ed1b7625 "", bool isTopLevel = true)+0x277 [c:\users\takushni\projects\other\xpiks-deps\exiv2-0.25\xmpsdk\src\parserdf.cpp @ 405]
> 00000091`b2efc9c0 00007ff8`ed02983e libexiv2!RDF_LiteralPropertyElement(class XMP_Node * xmpParent = 0x000001db`ce46e640, class XML_Node * xmlNode = 0x000001db`ce3d7f20, bool isTopLevel = true)+0x5f [c:\users\takushni\projects\other\xpiks-deps\exiv2-0.25\xmpsdk\src\parserdf.cpp @ 1015]
> 00000091`b2efcb10 00007ff8`ed0293f3 libexiv2!RDF_PropertyElement(class XMP_Node * xmpParent = 0x000001db`ce46e640, class XML_Node * xmlNode = 0x000001db`ce3d7f20, bool isTopLevel = true)+0x40e [c:\users\takushni\projects\other\xpiks-deps\exiv2-0.25\xmpsdk\src\parserdf.cpp @ 915]
> 00000091`b2efcc30 00007ff8`ed028fcc libexiv2!RDF_PropertyElementList(class XMP_Node * xmpParent = 0x000001db`ce46e640, class XML_Node * xmlParent = 0x000001db`ce394770, bool isTopLevel = true)+0x113 [c:\users\takushni\projects\other\xpiks-deps\exiv2-0.25\xmpsdk\src\parserdf.cpp @ 796]
> 00000091`b2efccd0 00007ff8`ed028eaf libexiv2!RDF_NodeElement(class XMP_Node * xmpParent = 0x000001db`ce46e640, class XML_Node * xmlNode = 0x000001db`ce394770, bool isTopLevel = true)+0xdc [c:\users\takushni\projects\other\xpiks-deps\exiv2-0.25\xmpsdk\src\parserdf.cpp @ 707]
> 00000091`b2efcd30 00007ff8`ed028dc6 libexiv2!RDF_NodeElementList(class XMP_Node * xmpParent = 0x000001db`ce46e640, class XML_Node * xmlParent = 0x000001db`ce484460, bool isTopLevel = true)+0xcf [c:\users\takushni\projects\other\xpiks-deps\exiv2-0.25\xmpsdk\src\parserdf.cpp @ 672]
> 00000091`b2efcdc0 00007ff8`ed028d43 libexiv2!RDF_RDF(class XMP_Node * xmpTree = 0x000001db`ce46e640, class XML_Node * xmlNode = 0x000001db`ce484460)+0x76 [c:\users\takushni\projects\other\xpiks-deps\exiv2-0.25\xmpsdk\src\parserdf.cpp @ 651]
> 00000091`b2efce00 00007ff8`ed00314d libexiv2!ProcessRDF(class XMP_Node * xmpTree = 0x000001db`ce46e640, class XML_Node * rdfNode = 0x000001db`ce484460, unsigned long options = 0)+0x43 [c:\users\takushni\projects\other\xpiks-deps\exiv2-0.25\xmpsdk\src\parserdf.cpp @ 629]
> 00000091`b2efce30 00007ff8`ecfcb755 libexiv2!XMPMeta::ParseFromBuffer(char * buffer = 0x000001db`ce6e8090 "<?xpacket begin='???", unsigned long xmpSize = 0x7ca, unsigned long options = 0)+0x6bd [c:\users\takushni\projects\other\xpiks-deps\exiv2-0.25\xmpsdk\src\xmpmeta-parse.cpp @ 1256]
> 00000091`b2efd000 00007ff8`ecfbf320 libexiv2!WXMPMeta_ParseFromBuffer_1(struct __XMPMeta__ * xmpRef = 0x000001db`ce46e630, char * buffer = 0x000001db`ce6e8090 "<?xpacket begin='???", unsigned long bufferSize = 0x7ca, unsigned long options = 0, struct WXMP_Result * wResult = 0x00000091`b2efd0d8)+0xb5 [c:\users\takushni\projects\other\xpiks-deps\exiv2-0.25\xmpsdk\src\wxmpmeta.cpp @ 1274]
> 00000091`b2efd0a0 00007ff8`ecfb99e5 libexiv2!TXMPMeta<std::basic_string<char,std::char_traits<char>,std::allocator<char> > >::ParseFromBuffer(char * buffer = 0x000001db`ce6e8090 "<?xpacket begin='???", unsigned long bufferSize = 0x7ca, unsigned long options = 0)+0x70 [c:\users\takushni\projects\other\xpiks-deps\exiv2-0.25\xmpsdk\include\client-glue\txmpmeta.incl_cpp @ 903]
> 00000091`b2efd130 00007ff8`ecfb0f1d libexiv2!TXMPMeta<std::basic_string<char,std::char_traits<char>,std::allocator<char> > >::TXMPMeta<std::basic_string<char,std::char_traits<char>,std::allocator<char> > >(char * buffer = 0x000001db`ce6e8090 "<?xpacket begin='???", unsigned long xmpSize = 0x7ca)+0x65 [c:\users\takushni\projects\other\xpiks-deps\exiv2-0.25\xmpsdk\include\client-glue\txmpmeta.incl_cpp @ 169]
> 00000091`b2efd170 00007ff8`ecec8c12 libexiv2!Exiv2::XmpParser::decode(class Exiv2::XmpData * xmpData = 0x000001db`ce35e458, class std::basic_string<char,std::char_traits<char>,std::allocator<char> > * xmpPacket = 0x000001db`ce35e4a0)+0x16d [c:\users\takushni\projects\other\xpiks-deps\exiv2-0.25\src\xmp.cpp @ 492]
> 00000091`b2efdf60 00007ff7`e6985e8a libexiv2!Exiv2::JpegBase::readMetadata(void)+0x942 [c:\users\takushni\projects\other\xpiks-deps\exiv2-0.25\src\jpgimage.cpp @ 390]
> 00000091`b2eff1a0 00007ff7`e6985baf xpiks_qt!MetadataIO::Exiv2ReadingWorker::readMetadata(class Models::ArtworkMetadata * artwork = 0x000001db`ce2aec30, struct MetadataIO::ImportDataResult * importResult = 0x00000091`b2eff470)+0x15a [c:\users\takushni\projects\other\xpiks\src\xpiks-qt\metadataio\exiv2readingworker.cpp @ 567]
> 00000091`b2eff3a0 00007ff7`e69ddd85 xpiks_qt!MetadataIO::Exiv2ReadingWorker::process(void)+0x15f [c:\users\takushni\projects\other\xpiks\src\xpiks-qt\metadataio\exiv2readingworker.cpp @ 532]
> 00000091`b2eff540 00000000`615b8592 xpiks_qt!MetadataIO::Exiv2ReadingWorker::qt_static_metacall(class QObject * _o = 0x000001db`ce341ae0, QMetaObject::Call _c = InvokeMetaMethod (0n0), int _id = 0n2, void ** _a = 0x00000091`b2eff610)+0x85 [c:\users\takushni\projects\other\xpiks\src\build-xpiks-qt-desktop_qt_5_6_0_msvc2015_64bit-debug\debug\moc_exiv2readingworker.cpp @ 86]
> 00000091`b2eff590 00000000`615b7d18 Qt5Cored!QMetaObject::activate(class QObject * sender = 0x000001db`ce34bd70, int signalOffset = 0n3, int local_signal_index = 0n0, void ** argv = 0x00000000`00000000)+0x862 [c:\users\qt\work\qt\qtbase\src\corelib\kernel\qobject.cpp @ 3732]
> 00000091`b2eff770 00000000`616c3394 Qt5Cored!QMetaObject::activate(class QObject * sender = 0x000001db`ce34bd70, struct QMetaObject * m = 0x00000000`61ab4be0, int local_signal_index = 0n0, void ** argv = 0x00000000`00000000)+0x38 [c:\users\qt\work\qt\qtbase\src\corelib\kernel\qobject.cpp @ 3596]
> 00000091`b2eff7a0 00000000`611ceadf Qt5Cored!QThread::started(struct QThread::QPrivateSignal __formal = struct QThread::QPrivateSignal)+0x24 [c:\users\qt\work\qt\qtbase\src\corelib\.moc\debug\moc_qthread.cpp @ 157]
> 00000091`b2eff7d0 00007ff8`f407a1bc Qt5Cored!QThreadPrivate::start(void * arg = 0x000001db`ce34bd70)+0x17f [c:\users\qt\work\qt\qtbase\src\corelib\thread\qthread_win.cpp @ 390]
> 00000091`b2eff860 00007ff8`f4079e13 ucrtbased!__crt_unique_heap_ptr<char const ,__crt_internal_free_policy>::detach+0x5c
> 00000091`b2eff8a0 00007ff9`2e208102 ucrtbased!register_onexit_function+0xd3
> 00000091`b2eff8f0 00007ff9`30cbc5b4 KERNEL32!BaseThreadInitThunk+0x22
> 00000091`b2eff920 00000000`00000000 ntdll!RtlUserThreadStart+0x34


```

**Comment by Ribtoks** [comment]
Weak attempt to fix that 22e9f4e

**Comment by Ribtoks** [comment]
Currently closing because exiv2 is not used anymore
